tests/scripts/run-corefx-tests.py

```diff
@@ -361,12 +361,32 @@ def main(args):
 
     # If we're doing altjit testing, then don't run any tests that don't work with altjit.
     if ci_arch is not None and (ci_arch == 'x86_arm_altjit' or ci_arch == 'x64_arm64_altjit'):
-        # The property value we need to specify is a semicolon separated list of two values,
-        # so the two values must be enclosed in double quotes. Without the quotes, msbuild
-        # thinks the item after the semicolon is a different named property. Also, the double
-        # quotes need preceeding backslashes or else run.exe (invoked from build-tests.cmd)
-        # will eat them. They need to be double backslashes so Python preserves the backslashes.
-        without_categories = '/p:WithoutCategories=\\"IgnoreForCI;XsltcExeRequired\\"'
+        # The property value we need to specify for the WithoutCategories property is a semicolon
+        # separated list of two values, so the two values must be enclosed in double quotes, namely:
+        #
+        #  /p:WithoutCategories="IgnoreForCI;XsltcExeRequired"
+        #
+        # Without the quotes, msbuild treats the semicolon as separating multiple name/value pairs,
+        # e.g.,
+        #
+        #  /p:name1=value1;name2=value2
+        #
+        # However, corefx invokes msbuild via "dotnet msbuild", and dotnet.exe itself parses and
+        # removes the quotes, so we need to force them to stay, by adding preceding backslashes:
+        #
+        #  /p:WithoutCategories=\"IgnoreForCI;XsltcExeRequired\"
+        #
+        # See, for instance, https://github.com/dotnet/cli/issues/7791.
+        #
+        # But we don't invoke "dotnet msbuild" directly; we invoke it via build-tests.cmd, which
+        # invokes run.cmd, which invokes run.exe. These also strip off backslashes. So, we need
+        # to pass the following to build-tests.cmd:
+        #
+        #  /p:WithoutCategories=\\\"IgnoreForCI;XsltcExeRequired\\\"
+        #
+        # Finally, Python treats backslashes as special, so we need to backslash-escape each
+        # backslash we wish to pass through, leading to six (!) backslashes front and back!
+        without_categories = '/p:WithoutCategories=\\\\\\"IgnoreForCI;XsltcExeRequired\\\\\\"'
```

Comment:
Oh this looks horrible and destined to be broken again. Did you look into another away to escape or encode `;`? Or perhaps we setup some other property we can set to avoid this issue.

Reply:
> Oh this looks horrible and destined to be broken again.

Agreed

I couldn't find any example of a different way to encode it.

I considered adding a special "-AltjitExclusions" option to run's config.json, but I'm not sure it makes it any/much better:

https://github.com/dotnet/corefx/compare/master...BruceForstall:AddAltjitConfigOption

Another option might be to add a set of properties, e.g., WithoutCategory1, WithoutCategory2, etc., and have the msbuild concatenate them as necessary.

I think the semicolon is only neede because it's treated as an msbuild array?

My msbuild wizardry is severely limited, so I didn't go down that route (and it would probably require changing buildtools... ugh)

Reply:
One option you could consider is to have yoru script create a response file and we simply pass that which would mostly eliminate all the special escaping through the stack.